geode-core/src/test/java/org/apache/geode/internal/statistics/LinuxSystemStatsTest.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.internal.statistics;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.geode.CancelCriterion;
+import org.apache.geode.Statistics;
+import org.apache.geode.internal.statistics.platform.LinuxProcFsStatistics;
+import org.apache.geode.internal.statistics.platform.LinuxSystemStats;
+import org.apache.geode.test.junit.categories.IntegrationTest;
+import org.apache.tools.ant.filters.StringInputStream;
+import org.junit.*;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+
+/**
+ * Technically a linux only test - the file handling is all mocked up so the test can run on any
+ * host os.
+ */
+@Category(IntegrationTest.class)
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore("*.IntegrationTest")
+@PrepareForTest(LinuxProcFsStatistics.class)
+@Ignore
+public class LinuxSystemStatsTest extends StatSamplerTestCase {
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private int[] ints;
+  private long[] longs;
+  private double[] doubles;
+  private LocalStatisticsFactory statisticsFactory;
+  private File testDir;
+
+  @Before
+  public void setUp() throws Exception {
+    this.testDir = this.temporaryFolder.getRoot();
+    assertTrue(this.testDir.exists());
+    System.setProperty(SimpleStatSampler.ARCHIVE_FILE_NAME_PROPERTY, this.testDir.getAbsolutePath()
```

Comment:
Since you're setting a System property, I recommend adding this rule:
```java
@Rule
public RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
```
If we change integrationTest to not fork JVMs, then this will make your test play nicely with other IntegrationTests. We already have a bunch that we'll need to change in order to avoid forking.

Reply:
Done.